src/components/GraphiQL.js

```diff
@@ -694,6 +710,42 @@ export class GraphiQL extends React.Component {
     editor.setValue(print(parse(editor.getValue())));
   };
 
+  handleDefragQuery = () => {
+    const editor = this.getQueryEditor();
+    const ast = parse(editor.getValue());
+
+    let operations = ast.definitions.filter(elem => {
+      return elem.kind !== 'FragmentDefinition';
+    });
+
+    const fragments = {};
+    ast.definitions
+      .filter(elem => {
+        return elem.kind === 'FragmentDefinition';
+      })
+      .forEach(frag => {
+        frag.kind = 'InlineFragment';
+        fragments[frag.name.value] = frag;
+      });
+    operations = operations.map(op => this.resolveDefinition(fragments, op));
+
+    ast.definitions = operations;
```

Comment:
I'd be more comfortable if we don't mutate `ast` - let's just create a new GraphQL ast tree from operations and print the value.